packages/connect-thegraph/src/connector.ts

```diff
@@ -39,58 +36,60 @@ export default class ConnectorTheGraph extends GraphQLWrapper
   }
 
   async rolesForAddress(appAddress: string): Promise<Role[]> {
-    const result = await this.performQuery(queries.ROLE_BY_APP_ADDRESS, {
-      appAddress: appAddress.toLowerCase(),
-    })
-
-    const datas = this.parseQueryResult(parseRoles, result)
-
-    return datas.map((data: RoleData) => {
-      return new Role(data, this)
-    })
+    return this.performQueryWithParser(
+      queries.ROLE_BY_APP_ADDRESS('query'),
+      { appAddress: appAddress.toLowerCase() },
+      parseRoles
+    )
   }
 
   async permissionsForOrg(orgAddress: string): Promise<Permission[]> {
-    const result = await this.performQuery(queries.ORGANIZATION_PERMISSIONS, {
-      orgAddress: orgAddress.toLowerCase(),
-    })
-
-    const datas = this.parseQueryResult(parsePermissions, result)
+    return this.performQueryWithParser(
+      queries.ORGANIZATION_PERMISSIONS('query'),
+      { orgAddress: orgAddress.toLowerCase() },
+      parsePermissions
+    )
+  }
 
-    return datas.map((data: PermissionData) => {
-      return new Permission(data, this)
-    })
```

Comment:
And `subscribeToQueryWithParser(...)` is `performQueryWithParser(...)`'s analogy function with subscriptions. With this, the connector looks very straight forward and is easy to digest.